timesketch/lib/tasks.py

```diff
@@ -181,37 +186,78 @@ def build_index_pipeline(file_path, timeline_name, index_name, file_extension,
     return chain(index_task, index_analyzer_chain)
 
 
-def build_sketch_analysis_pipeline(sketch_id):
+def build_sketch_analysis_pipeline(sketch_id, searchindex_id, user_id,
+                                   analyzer_names=None, analyzer_kwargs=None):
     """Build a pipeline for sketch analysis.
 
     Args:
-        sketch_id: The ID of the sketch to analyze.
+        sketch_id (int): The ID of the sketch to analyze.
+        searchindex_id (int): The ID of the searchindex to analyze.
+        user_id (int): The ID of the user who started the analyzer.
+        analyzer_names (list): List of analyzers to run.
+        analyzer_kwargs (dict): Arguments to the analyzers.
 
     Returns:
         Celery group with analysis tasks or None if no analyzers are enabled.
     """
     tasks = []
+    auto_analyzers = current_app.config.get('AUTO_SKETCH_ANALYZERS', None)
 
-    # Exit early if sketch analyzers are disabled.
-    if not current_app.config.get('ENABLE_SKETCH_ANALYZERS', False):
+    # Exit early if no sketch analyzers are configured to run.
+    if not (analyzer_names or auto_analyzers):
         return None
 
-    for analyzer_name, analyzer_cls in manager.AnalysisManager.get_analyzers():
-        kwarg_list = analyzer_cls.get_kwargs()
+    if not analyzer_names:
+        analyzer_names = auto_analyzers
```

Comment:
ok... so you are forgetting to actually use auto_analyzers except in the case where there are no analyzers defined...

should this not be

```
auto_analyzers = current....get('AUTO+SKETCH_ANALYZERS', [])
...
if analyzer_names:
  analyzer_names.extend(auto_analyzers)
else:
  analyzer_names = auto_analyzers
```

That is to extend the analyzer names with the auto ones? or is this meant to be called twice, once in the beginning where we have just the auto analyzers and then a potentially second/third/etc times when it is manually run? 
if that is the case I would argue that this should be

```
if not analyzer_names:
  analyzer_names = current_app.config.get('AUTO_SKETCH_ANALYZERS', []

if not analyzer_names:
  return
```

and just not define the `auto_analyzers` attribute at all

Reply:
Good catch, it is the latter. Changed per your suggestion which is clearer.